src/Jira/Api.php

```diff
@@ -875,14 +882,23 @@ public function getProjectIssueTypes($project_key)
 	/**
 	 * Returns a list of all resolutions.
 	 *
-	 * @param string $project_key Project key.
-	 *
-	 * @return array|false
+	 * @return array
 	 * @since  2.0.0
 	 */
 	public function getResolutions()
 	{
-		return $this->api(self::REQUEST_GET, '/rest/api/2/resolution', array(), true);
+		if ( !count($this->resolutions) ) {
```

Comment:
Agreed that it is a bit wonky now. Refactored it to use null when the list is not fetched, and doa  type safe check on that. 

This way we can make a distinction between non-fetched and empty list. (Internally only, consumers always get an array returned when using the method)